march_hardware/src/Encoder.cpp

```diff
@@ -78,6 +78,26 @@ bool Encoder::isWithinSoftLimitsIU(int positionIU)
   return (positionIU > this->lowerSoftLimitIU && positionIU < this->upperSoftLimitIU);
 }
 
+bool Encoder::isValidTargetIU(int currentIU, int targetIU)
+{
+  if (this->isWithinSoftLimitsIU(targetIU))
+  {
+    return true;
+  }
+
+  if (currentIU >= this->getUpperSoftLimitIU())
+  {
+    return (targetIU <= currentIU) && (targetIU > this->getLowerSoftLimitIU());
+  }
+
+  if (currentIU <= this->getLowerSoftLimitIU())
+  {
+    return (targetIU >= currentIU) && (targetIU < this->getUpperSoftLimitIU());
+  }
+
```

Comment:
im not sure if this is a relevant detail but should the order of these if statements not be different? With the first one actually being at the bottom? In my mind it makes sense to always first check the current IU with respect to the limits and only once it within them to allow for unrestricted movement within soft limits

Reply:
Why? If the target is within limits, movements is always allowed

Reply:
@Bart-van-Ingen ping

Reply:
@Bart-van-Ingen ping

Reply:
rereading this I see that it does make sense